c++/src/capnp/compat/json.c++

```diff
@@ -379,13 +382,208 @@ void JsonCodec::encodeField(StructSchema::Field field, DynamicValue::Reader inpu
   encode(input, field.getType(), output);
 }
 
+namespace {
+int64_t parseInt64(kj::StringPtr s) {
+  char *endPtr;
+  errno = 0;
+  int64_t value = std::strtoll(s.begin(), &endPtr, 10);
+  KJ_REQUIRE(endPtr == s.end(), "String does not contain valid number", s);
+  KJ_REQUIRE(errno != ERANGE, "Value out-of-range", s);
```

Comment:
One thing I'm not sure of: is it best to check for the error return as well as `errno`? In this case, that'd be checking if it was `LLONG_MIN` or `LLONG_MAX`. The general protocol for `errno` is to ignore it unless the return value indicated an error.